File: scripts/capture-stats.py
import asyncio
from playwright.async_api import async_playwright
import os

async def capture():
    async with async_playwright() as p:
        browser = await p.chromium.launch(
            headless=True,
            args=['--no-sandbox', '--disable-setuid-sandbox', '--disable-dev-shm-usage', '--disable-gpu', '--single-process']
        )
        page = await browser.new_page()
        await page.set_viewport_size({"width": 1440, "height": 900})

        try:
            print('Navigating to page...')
            await page.goto('http://127.0.0.1:3000', wait_until='domcontentloaded', timeout=30000)
            
            print('Waiting for content...')
            await page.wait_for_selector('img[alt="Sparko Trophy"]', timeout=20000)
            
            print('Found trophy, taking screenshots...')
            
            # Take a full viewport screenshot first
            await page.screenshot(path='/home/z/my-project/download/stats-full-viewport.png', full_page=False)
            print('Full viewport screenshot saved!')
            
            # Scroll to find the stats section
            # Find the bounding box of the trophy section
            trophy_box = await page.evaluate('''() => {
                const trophy = document.querySelector('img[alt="Sparko Trophy"]');
                if (!trophy) return null;
                let parent = trophy.parentElement;
                while (parent && parent.tagName !== 'SECTION') {
                    parent = parent.parentElement;
                }
                if (!parent) return null;
                const rect = parent.getBoundingClientRect();
                return { x: rect.x, y: rect.y, width: rect.width, height: rect.height, top: rect.top };
            }''')
            
            print(f'Section box: {trophy_box}')
            
            if trophy_box:
                # Scroll to the section
                await page.evaluate(f'window.scrollTo(0, {trophy_box["top"] - 100})')
                await asyncio.sleep(0.5)
                
                # Clip screenshot to just the section area
                await page.screenshot(
                    path='/home/z/my-project/download/stats-section-current.png',
                    clip={
                        'x': trophy_box['x'],
                        'y': trophy_box['y'],
                        'width': trophy_box['width'],
                        'height': trophy_box['height']
                    }
                )
                print('StatsSection clipped screenshot saved!')

        except Exception as err:
            print(f'Error: {err}')
            import traceback
            traceback.print_exc()

        await browser.close()
        print('Done')

asyncio.run(capture())

File: scripts/capture-stats.js
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch({
    headless: 'new',
    args: ['--no-sandbox', '--disable-setuid-sandbox', '--disable-dev-shm-usage', '--disable-gpu', '--single-process', '--disable-extensions'],
  });
  const page = await browser.newPage();
  await page.setViewport({ width: 1440, height: 900, deviceScaleFactor: 2 });

  try {
    await page.goto('http://localhost:3000', { waitUntil: 'networkidle0', timeout: 30000 });

    // Wait for the StatsSection to render
    await page.waitForSelector('img[alt="Sparko Trophy"]', { timeout: 15000 });

    // Find the section containing the trophy
    const section = await page.evaluateHandle(() => {
      const trophy = document.querySelector('img[alt="Sparko Trophy"]');
      if (!trophy) return null;
      let parent = trophy.parentElement;
      while (parent && parent.tagName !== 'SECTION') {
        parent = parent.parentElement;
      }
      return parent;
    });

    if (section && section.asElement()) {
      await section.asElement().screenshot({ path: '/home/z/my-project/download/stats-section-current.png', type: 'png' });
      console.log('StatsSection screenshot saved!');
    }

    // Also take a wider screenshot showing the full section area
    await page.evaluate(() => {
      window.scrollTo(0, document.body.scrollHeight * 0.4); // scroll to middle of page
    });
    await new Promise(r => setTimeout(r, 500));

    // Full viewport screenshot
    await page.screenshot({ path: '/home/z/my-project/download/full-page-current.png', type: 'png', fullPage: false });
    console.log('Full page screenshot saved!');

  } catch (err) {
    console.error('Error:', err.message);
  }

  await browser.close();
  console.log('Done');
})();
